protocol/persistence.go

```diff
@@ -498,6 +501,18 @@ func (db sqlitePersistence) Chat(chatID string) (*Chat, error) {
 		if profile.Valid {
 			chat.Profile = profile.String
 		}
+
+		// Set UnviewedCounts and make sure they are above 0
+		// Since Chat's UnviewedMessagesCount is uint and the SQL column is INT, it can create a discrepancy
+		if unviewedMessagesCount < 0 {
+			unviewedMessagesCount = 0
+		}
+		if unviewedMentionsCount < 0 {
+			unviewedMentionsCount = 0
+		}
+		chat.UnviewedMessagesCount = uint(unviewedMessagesCount)
+		chat.UnviewedMentionsCount = uint(unviewedMentionsCount)
```

Comment:
The checks above should indeed not be necessary, because judging by the original issue, SCAN would've failed already at that point and we wouldn't even make it into this code path.